packages/x-data-grid-pro/src/tests/editComponents.DataGridPro.test.tsx

```diff
@@ -135,10 +135,9 @@ describe('<DataGridPro /> - Edit components', () => {
 
       const input = within(cell).getByRole<HTMLInputElement>('textbox');
       fireEvent.change(input, { target: { value: 'Puma' } });
-      await act(() => Promise.resolve());
-
       expect(onValueChange.callCount).to.equal(1);
       expect(onValueChange.lastCall.args[1]).to.equal('Puma');
+      await act(() => Promise.resolve());
```

Comment:
This feels somewhat hacky with an "empty" `act` at the end. 🤔 
WDYT about wrapping the first expect in a `waitFor`? 🤔 
```suggestion
      await waitFor(() => expect(onValueChange.callCount).to.equal(1));
      expect(onValueChange.lastCall.args[1]).to.equal('Puma');
```

Reply:
I haven't removed the `await act(() => Promise.resolve());` in case there is some clean up to do before  going to the next test? We can do this too:

```suggestion
      expect(onValueChange.callCount).to.equal(1);
      expect(onValueChange.lastCall.args[1]).to.equal('Puma');
```

would looks much better.